### PR TITLE
Testing boilerplate

### PR DESCRIPTION
**Pull request by robbiepitts**
- [x] Add appropriate files to `files/`
- [x] Publish `glimmer-vm`
- [x] Land https://github.com/glimmerjs/glimmer-application-pipeline/pull/109
- [x] Land https://github.com/glimmerjs/glimmer-test-helpers/pull/1
- [x] Release `@glimmer/test-helpers`
- [x] Land https://github.com/glimmerjs/glimmer-application-pipeline/pull/114
- [x] Release `@glimmer/application-pipeline`
- [x] Bump `@glimmer/application-pipeline`
- [ ] Land this
- [ ] Release `@glimmer/blueprint`

### CODE REVIEW

**Comment by billheaton-q2** [comment]
@robbiepitts what do you think about extracting the ember-native-dom-helpers code for Glimmer Testing (as a TS library) ?

See https://github.com/cibernox/ember-native-dom-helpers/tree/master/addon-test-support

**Comment by pixelhandler** [comment]
@robbiepitts what do you think about extracting the ember-native-dom-helpers code for Glimmer Testing (as a TS library) ?

See https://github.com/cibernox/ember-native-dom-helpers/tree/master/addon-test-support

**Comment by robbiepitts** [comment]
@pixelhandler I think it would be a very worth endeavor. I'm sure we can come up with an API that is framework-agnostic, or at least something that can be used between Ember and Glimmer.

**Comment by pixelhandler** [comment]
@cibernox @rwjblue - perhaps the path to using TypeScript for ember-native-dom-helpers is to create a library for the test helper code which can be shared between Glimmer and Ember. Maybe ember-native-dom-helpers can import a TypeScript library for it's test helpers and provide them as an addon to consuming Ember apps. Likewise, Glimmer's testing solution could use the same dependency for test helpers.

**Comment by cibernox** [comment]
@pixelhandler There is a loose end there still. While `find` and `findAll` are easy to extract, "action" helpers like `click` and `fillIn` return `wait()`, which comes from 'ember-test-helpers'.

IIRC, `wait` waits for the runloop to be empty, ajax requests to be settled and custom timers (if any) to be finished. Unless I am mistaken, glimmer also some sort of runloop, but it would take some work to create the equivalent of `wait` for glimmer. Having such helper, extracting the non-ember specific helper to another library would be easy.

**Comment by rwjblue** [comment]
We need config to allow a "wait" helper to be provided to us.  glimmer apps will have the same thing available ("wait for rendering to be completed").

If we can allow the function to be provided, then we can start making this be a bit more abstracted...

**Comment by cibernox** [comment]
@rwjblue are you suggesting that wait should live in ember and glimmer?

**Comment by robbiepitts** [comment]
@cibernox In Ember and Glimmer you are waiting for different things, so the idea is that you provide to the helper library an interface that affords waiting in a generic sense. So yes, wait would be both in Ember and Glimmer.

**Comment by cibernox** [comment]
@robbiepitts I can help with that perhaps. @rwjblue would this require an RFC or should be part of the grand testing unification?

File: files/src/ui/components/__component__/component-test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { setupRenderingTest } from '@glimmer/test-helpers';
+import hbs from '@glimmer/inline-precompile';
+
+const { module, test } = QUnit;
+
+module('Component: <%= component %>', function(hooks) {
+  setupRenderingTest(hooks);
```
Comment (cibernox) [inline files/src/ui/components/__component__/component-test.ts]: This has been merged already, but `hooks` will be undefined, so this blueprint is going to generate a test that has a syntax error.

File: files/src/utils/test-helpers/test-helper.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { setApp, start } from '@glimmer/test-helpers';
+import App from '../../main';
+
+QUnit.config.autostart = false;
```
Comment (cibernox) [inline files/src/utils/test-helpers/test-helper.ts]: Can't `Qunit` be imported instead of used as a global?

Comment (robbiepitts) [inline files/src/utils/test-helpers/test-helper.ts]: Not afaik.

File: files/src/ui/components/__component__/component-test.ts (hunk 1 above)
Comment (robbiepitts) [inline files/src/ui/components/__component__/component-test.ts]: Why will hooks be undefined?

Comment (cibernox) [inline files/src/ui/components/__component__/component-test.ts]: Darn, I didn't see it was passed as an argument. Nevermind

File: files/src/utils/test-helpers/test-helper.ts (hunk 2 above)
Comment (knownasilya) [inline files/src/utils/test-helpers/test-helper.ts]: That would be nice. cc @trentmwillis

Comment (trentmwillis) [inline files/src/utils/test-helpers/test-helper.ts]: QUnit gets defined as an [AMD module](https://github.com/qunitjs/qunit/blob/35d960bb2c2b52662264f9c7698ee38f7af41df6/src/export.js#L31-L33), but it's anonymous. Maybe we could leverage [this](https://ember-cli.com/user-guide/#standard-anonymous-amd-asset)?

Comment (cibernox) [inline files/src/utils/test-helpers/test-helper.ts]: I'm pretty sure we can if we explicitly use that in the `ember-cli-build.js`, but can we make that happen automatically? I suppose we'd have to create our own npm package wrapping npmjs, which is a shame.

Comment (trentmwillis) [inline files/src/utils/test-helpers/test-helper.ts]: It won't be changed in QUnit itself (for background, see https://github.com/qunitjs/qunit/pull/971#issuecomment-201704792). Don't we need some sort of wrapper/setup to include QUnit in the build anyway?

Comment (cibernox) [inline files/src/utils/test-helpers/test-helper.ts]: How would we name that? `ember-cli-qunit` is technically correct, as it hooks into ember-cli, but it currently contains ember-specific stuff.

Comment (rwjblue) [inline files/src/utils/test-helpers/test-helper.ts]: Making fake modules is a bit harder for us here, due to the fact that we are embracing "standard JavaScript" things and all modules need to be resolvable in node_modules (or we get annoying typescript and rollup errors...

**Pull request merged**